apps/auxo-app/src/hooks/useTransactionHandler.ts

```diff
@@ -1,9 +1,55 @@
 import { AnyAction } from "@reduxjs/toolkit";
-import { ContractTransaction } from "ethers";
+import { ContractTransaction, ethers } from "ethers";
 import { Dispatch, useEffect, useRef, useState } from "react";
 import { useAppDispatch, useAppSelector } from ".";
 import { setAlert } from "../store/app/app.slice";
-import { checkForEvent } from "../utils/event";
+
+/**
+ * Awaits the transaction confirmation to see if an event is fired
+ * @param tx the transaction
+ * @param eventName string name of the event
+ * @returns whether or not the event was fired
+ */
+export const checkForEvent = async (
+  tx: ContractTransaction,
+  eventName: string
+): Promise<boolean> => {
+  const awaitedTx = await tx.wait();
```

Comment:
No error handling on this promise.

Reply:
I think this speaks to the original point you made - the error handling is being done in the component which is far less clean than just having a single handler

Reply:
imo error handling should be done on data sourcing and errors should bubble up to components.. in that way you can change the way components handle errors or even make other custom components